src/lib/support/tests/TestZclString.cpp

```diff
@@ -0,0 +1,166 @@
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a unit test suite for ZclString
+ *
+ */
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <nlunit-test.h>
+#include <support/CHIPMem.h>
+#include <support/CodeUtils.h>
+#include <support/UnitTestRegistration.h>
+
+#include <lib/support/ZclString.h>
+
+using namespace chip;
+using namespace chip::Logging;
+using namespace chip::Platform;
+
+static void TestZclString64Chars(nlTestSuite * inSuite, void * inContext)
+{
+    // Init zcl string
+    uint8_t zclString[255];
+    // Init c string
+    char * cString64 = static_cast<char *>(MemoryCalloc(1024, true));
+    // Add random chars to c string
+    for (size_t i = 0; i < 64; i++)
+        cString64[i] = 'A';
+
+    MakeZclCharString(zclString, cString64);
+    int charIsTerminated = 0;
+    for (int i = 0; i != 256; i++)
+        if (zclString[i] == '\0')
+        {
+            charIsTerminated = 1;
+            break;
+        }
+
+    // Memmory assert
+    NL_TEST_ASSERT(inSuite, cString64 != nullptr);
+    // ZCL String has char terminated
+    NL_TEST_ASSERT(inSuite, charIsTerminated == 1);
+    // ZCL String length assert
+    NL_TEST_ASSERT(inSuite, zclString[0] == 64);
+
+    chip::Platform::MemoryFree(cString64);
+}
+
+static void TestZclString254Chars(nlTestSuite * inSuite, void * inContext)
+{
+    // Init zcl string
+    uint8_t zclString[255];
+    // Init c string
+    char * cString254 = static_cast<char *>(MemoryCalloc(1024, true));
+    // Add random chars to c string
+    for (size_t i = 0; i < 254; i++)
+        cString254[i] = 'A';
+
+    MakeZclCharString(zclString, cString254);
+    int charIsTerminated = 0;
+    for (int i = 0; i != 256; i++)
+        if (zclString[i] == '\0')
+        {
+            charIsTerminated = 1;
+            break;
+        }
+
+    // Memmory assert
+    NL_TEST_ASSERT(inSuite, cString254 != nullptr);
+    // ZCL String has char terminated
+    NL_TEST_ASSERT(inSuite, charIsTerminated == 1);
+    // ZCL String length assert
+    NL_TEST_ASSERT(inSuite, zclString[0] == 254);
+
+    chip::Platform::MemoryFree(cString254);
+}
+
+static void TestZclString255Chars(nlTestSuite * inSuite, void * inContext)
+{
+    // Init zcl string
+    uint8_t zclString[255];
+    // Init c string
+    char * cString255 = static_cast<char *>(MemoryCalloc(1024, true));
+    // Add random chars to c string
+    for (size_t i = 0; i < 255; i++)
+        cString255[i] = 'A';
+
+    MakeZclCharString(zclString, cString255);
+    int charIsTerminated = 0;
+    for (int i = 0; i != 256; i++)
+        if (zclString[i] == '\0')
+        {
+            charIsTerminated = 1;
+            break;
+        }
+
+    // Memmory assert
+    NL_TEST_ASSERT(inSuite, cString255 != nullptr);
+    // ZCL String has char terminated
+    NL_TEST_ASSERT(inSuite, charIsTerminated == 1);
```

Comment:
This will test true because it will treat the 0 byte for the length as a "terminator", but I have no idea what this is even testing for....

Reply:
Removed these tests.